### ISSUE TITLE
Release 0.23.1

### ISSUE THREAD

**Issue by zsxwing** [ReactiveX/RxScala#104]
Can we release 0.23.1 now, or is there any TODO before releasing it? Here is the release note: #103.

/cc @headinthebox @samuelgruetter @benjchristensen

**Comment by samuelgruetter** [comment]
I think we can

**Comment by benjchristensen** [comment]
:+1:

**Comment by zsxwing** [comment]
@headinthebox is there other non breaking change you want for the Coursera MOOC?

**Comment by headinthebox** [comment]
Give me a few days (after Christmas) to port all the code to the latest version.

**Comment by zsxwing** [comment]
> Give me a few days (after Christmas) to port all the code to the latest version.

I guess you are using 0.23.0. Do you need 0.23.1, especially the fix for `Subscription.isUnsubscribed`? If so, we can release it now and fix the further issues in 0.23.2.

**Comment by zsxwing** [comment]
@headinthebox any update?

**Comment by benjchristensen** [comment]
@headinthebox may be slow to respond right now so you might need to move forward without confirmation from him.

**Comment by headinthebox** [comment]
Yeah, go ahead.

**Comment by zsxwing** [comment]
OK. I will release 0.23.1 today.

**Comment by zsxwing** [comment]
Done.

**Comment by zsxwing** [comment]
My mistake... :( A typo in the commit message of https://github.com/ReactiveX/RxScala/commit/1da79a0ac386c5d3ddde7f9774f8006cef978b31
But users won't be affected.

**Comment by zsxwing** [comment]
publishViaTravis.sh failed:

```
[info]  published rxscala_2.10 to https://api.bintray.com/maven/reactivex/RxJava/RxJava/io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1.jar
[info]  published rxscala_2.10 to https://api.bintray.com/maven/reactivex/RxJava/RxJava/io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1-javadoc.jar
{"message":"Unable to upload files: An artifact with the path 'io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1-javadoc.jar.asc' already exists"}
java.lang.RuntimeException: error uploading to https://api.bintray.com/maven/reactivex/RxJava/RxJava/io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1-javadoc.jar.asc: {"message":"Unable to upload files: An artifact with the path 'io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1-javadoc.jar.asc' already exists"}
    at bintray.BintrayMavenRepository.put(Resolver.scala:25)
    at org.apache.ivy.plugins.resolver.RepositoryResolver.put(RepositoryResolver.java:234)
    at org.apache.ivy.plugins.resolver.RepositoryResolver.publish(RepositoryResolver.java:216)
    at sbt.IvyActions$$anonfun$publish$3.apply(IvyActions.scala:287)
    at sbt.IvyActions$$anonfun$publish$3.apply(IvyActions.scala:286)
    at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:772)
    at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
    at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
    at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:771)
    at sbt.IvyActions$.publish(IvyActions.scala:286)
    at sbt.IvyActions$$anonfun$publish$1$$anonfun$apply$1.apply$mcV$sp(IvyActions.scala:113)
    at sbt.IvyActions$$anonfun$publish$1$$anonfun$apply$1.apply(IvyActions.scala:113)
    at sbt.IvyActions$$anonfun$publish$1$$anonfun$apply$1.apply(IvyActions.scala:113)
    at sbt.IvyActions$.withChecksums(IvyActions.scala:122)
    at sbt.IvyActions$.sbt$IvyActions$$withChecksums(IvyActions.scala:117)
    at sbt.IvyActions$$anonfun$publish$1.apply(IvyActions.scala:113)
    at sbt.IvyActions$$anonfun$publish$1.apply(IvyActions.scala:106)
    at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:147)
    at sbt.IvySbt$Module$$anonfun$withModule$1.apply(Ivy.scala:147)
    at sbt.IvySbt$$anonfun$withIvy$1.apply(Ivy.scala:124)
    at sbt.IvySbt.sbt$IvySbt$$action$1(Ivy.scala:56)
    at sbt.IvySbt$$anon$3.call(Ivy.scala:64)
    at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:93)
    at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:78)
    at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:97)
    at xsbt.boot.Using$.withResource(Using.scala:10)
    at xsbt.boot.Using$.apply(Using.scala:9)
    at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:58)
    at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:48)
    at xsbt.boot.Locks$.apply0(Locks.scala:31)
    at xsbt.boot.Locks$.apply(Locks.scala:28)
    at sbt.IvySbt.withDefaultLogger(Ivy.scala:64)
    at sbt.IvySbt.withIvy(Ivy.scala:119)
    at sbt.IvySbt.withIvy(Ivy.scala:116)
    at sbt.IvySbt$Module.withModule(Ivy.scala:147)
    at sbt.IvyActions$.publish(IvyActions.scala:106)
    at sbt.Classpaths$$anonfun$publishTask$1.apply(Defaults.scala:1215)
    at sbt.Classpaths$$anonfun$publishTask$1.apply(Defaults.scala:1214)
    at scala.Function3$$anonfun$tupled$1.apply(Function3.scala:35)
    at scala.Function3$$anonfun$tupled$1.apply(Function3.scala:34)
    at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
    at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
    at sbt.std.Transform$$anon$4.work(System.scala:63)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
    at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
    at sbt.Execute.work(Execute.scala:235)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
    at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
    at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
[error] (root/*:publishSigned) error uploading to https://api.bintray.com/maven/reactivex/RxJava/RxJava/io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1-javadoc.jar.asc: {"message":"Unable to upload files: An artifact with the path 'io/reactivex/rxscala_2.10/0.23.1/rxscala_2.10-0.23.1-javadoc.jar.asc' already exists"}
[error] Total time: 42 s, completed Jan 16, 2015 2:20:16 AM
Upload failed.
./publishViaTravis.sh: line 24: return: can only `return' from a function or sourced script
```

**Comment by zsxwing** [comment]
https://travis-ci.org/ReactiveX/RxScala/builds/47198130

**Comment by zsxwing** [comment]
cc @jmhofer

**Comment by zsxwing** [comment]
@benjchristensen could you help delete https://bintray.com/reactivex/RxJava/RxScala/0.23.1/view/files/io/reactivex/rxscala_2.10/0.23.1 these files from Bintray so that I can make Travis CI republish it again?

**Comment by benjchristensen** [comment]
I deleted it. It had the 2.10 but not 2.11 versions of files.

**Comment by zsxwing** [comment]
@jmhofer I rebuilt it after @benjchristensen deleted the files. I checked Bintray and confirmed that there was no file in 0.23.1 before rebuilt it. But it still failed with the same error: https://travis-ci.org/ReactiveX/RxScala/builds/47198130

Have you encountered this issue before?

**Comment by zsxwing** [comment]
I can publish RxScala using my personal account manually: https://bintray.com/zsxwing/rxscalatest/RxScala/view/files/io/reactivex

Still have no idea where is the problem :(

**Comment by zsxwing** [comment]
@benjchristensen could you help release 0.23.1 and check it if the error will happen again?

**Comment by benjchristensen** [comment]
It is still happening. I deleted from Bintray and re-ran the build. 

I suggest either trying 0.23.2 or disabling Javadocs and see if those work?

**Comment by zsxwing** [comment]
@benjchristensen try 0.23.2 and the error is still happening: https://travis-ci.org/ReactiveX/RxScala/builds/47743317

Since you have the key, could you publish it in your local machine manually, so that we can know if it's an issue of Travis CI, or Bintray?

**Comment by zsxwing** [comment]
Finally, published 0.23.1 to Bintray successfully. It will be synchronized to Maven Central soon.

**Comment by zsxwing** [comment]
Close as RxScala 0.23.1 is available on Maven Central
